runtime/src/bank.rs

```diff
@@ -1525,8 +1525,10 @@ impl Bank {
         self.message_processor
             .add_instruction_processor(program_id, process_instruction);
 
-        // Register a bogus executable account, which will be loaded and ignored.
-        self.register_native_instruction_processor("", &program_id);
+        if self.get_account(&program_id).is_none() {
```

Comment:
How about deleting the existing account instead? Otherwise, you're still dealing with *its* loader.

Reply:
If you call this method, the native loader won't ever get called, the message processor will call `process_instruction` instead.

Reply:
I think what you're suggesting is to keep the current behaviour. Because without this added check, the existing account will get overwritten. I'm open to that, but I would like to have a way to add an instruction processor without changing the bank slot hash.

Reply:
I see now. I'll need to think about that. In the meantime, your PR problem description didn't describe your problem. :)

Reply:
Very true :) fixed

Reply:
Ok, I think your solution will work so long as the account's owner is the native loader. I'm not sure what would happen if it were, say, the BPF loader. You might see a one-off bug in the slice of accounts passed to the instruction processor.

Reply:
That's a really good point, need to think about that

Reply:
Well, you couldn't pass a bpf loader program instruction processor here actually. This [`ProcessInstruction`](https://github.com/solana-labs/solana/blob/master/sdk/src/entrypoint.rs#L15) != this [`ProcessInstruction`](https://github.com/solana-labs/solana/blob/master/runtime/src/message_processor.rs#L131)

Reply:
Should there be an assert that if the account exists, it should be owned by the native loader?

Reply:
I think it would definitely be a bug if you tried to add an instruction processor for an account that already exists *and* is not already a native loader program account

Reply:
An assert sounds good to me.

Reply:
Ok, added!